### PR TITLE
Re-fixing #6 and #23, resolving #36, #38, and #39

### PR DESCRIPTION
**Pull request by Iiridayn**

### CODE REVIEW

File: src/query-parser.js [hunk 1]
```diff
@@ -19,43 +19,51 @@ var QueryParser = function( provider ) {
...
  */
 QueryParser.prototype.createQuery = function( parsedInput ) {
-  var row,
+  var predicate,
     condition,
-    query = null,
+    query,
     i
 
+  query = rethink.table( parsedInput.table )
```
Comment (Iiridayn) [inline src/query-parser.js]: should be rethinkdb

File: src/query-parser.js [hunk 2]
```diff
@@ -19,43 +19,51 @@ var QueryParser = function( provider ) {
...
-    row = rethinkdb.row( condition[ 0 ] )[ condition[ 1 ] ]( condition[ 2 ] )
-
-    if( query === null ) {
-      query = row
+    if( condition[ 1 ] !== 'in' ) {
+      predicate = rethinkdb.row( condition[ 0 ] )[ condition[ 1 ] ]( condition[ 2 ] )
     } else {
-      query = query.and( row )
+      predicate = function( record ) {
+        return rethink.expr( condition[ 2 ] ).contains( record( condition[ 0 ] ) )
```
Comment (Iiridayn) [inline src/query-parser.js]: Also should be rethinkdb

**Comment by yasserf** [comment]
Hey @Iiridayn. I apologize for the late response merging this in, it runs so I merged it in, but would be great if you could add some tests for the extra scenarios you added. Thanks!

**Comment by Iiridayn** [comment]
Not sure how best to test #39 - race conditions are a bit odd :/. My fix didn't break anything though, which is evidence it at least returns correct data at the right time. Almost done with the rest.

**Pull request merged**